src/cueUtilities.js

```diff
@@ -319,60 +320,83 @@ module.exports = function (params, cy, api) {
-              && cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-              if(opts.undoable && !ur)
-                ur = cy.undoRedo({
-                  defaultActions: false
+          return null;
+        }
+
+        function cueClick(event) {
+          let node = hoveredGroup;
+          if (!node && selectedGroups.length > 0) {
+            node = getGroupOfClickedOnCue(event);
+          }
+          // no cue exists
+          if (!node) {
+            return;
+          }
+          let opts = options();
+
+          if (clickedOnCueRegion(node, event)) {
+            event.stopPropagation();
+            if(opts.undoable && !ur)
+              ur = cy.undoRedo({
+                defaultActions: false
+              });
+            if(api.isCollapsible(node))
+              if (opts.undoable){
+                ur.do("collapse", {
+                  nodes: node,
+                  options: opts
                 });
-              if(api.isCollapsible(node))
-                if (opts.undoable){
-                  ur.do("collapse", {
-                    nodes: node,
-                    options: opts
-                  });
-                }
-                else
-                  api.collapse(node, opts);
-              else if(api.isExpandable(node))
-                if (opts.undoable)
-                  ur.do("expand", {
-                    nodes: node,
-                    options: opts
-                  });
-                else
-                  api.expand(node, opts);
-
-              // don't draw the cue if the group is expandable, as mouse won't be inside it
-              if (hoveredGroup && !isSelectedGroupsContains(hoveredGroup) && api.isExpandable(node)) {
+              }
+              else
+                api.collapse(node, opts);
+            else if(api.isExpandable(node))
+              if (opts.undoable)
+                ur.do("expand", {
+                  nodes: node,
+                  options: opts
+                });
+              else
+                api.expand(node, opts);
+            if (options().appearOnGroupSelect) {
+              // mouse won't be in collapsed group, so shouldn't show cue
+              if (hoveredGroup && api.isExpandable(node)) {
                 hoveredGroup = null;
               }
               refreshCanvasImages();
             }
           }
-        });
+        }
+
+        function isClickOnAnyCueRegion(event) {
+          return (hoveredGroup && clickedOnCueRegion(hoveredGroup, event)) || (selectedGroups.length > 0 && getGroupOfClickedOnCue(event));
+        }
+
+        function stopEvent(event) {
```

Comment:
The name `stopEvent()` may be a bit broad. Maybe `stopCueEvent()` or `interceptCueEvent()`?